src/client/sandbox/xhr.js

```diff
@@ -73,6 +81,38 @@ export default class XhrSandbox extends SandboxBase {
             nativeMethods.xhrRemoveEventListener.call(this, 'readystatechange', syncCookieWithClient);
         };
 
+        const xmlHttpRequestWrapper = function () {
+            const xhr = new nativeMethods.XMLHttpRequest();
+
+            nativeMethods.xhrAddEventListener.call(xhr, 'readystatechange', emitXhrCompletedEvent);
+            nativeMethods.xhrAddEventListener.call(xhr, 'readystatechange', syncCookieWithClient);
+
+            return xhr;
+        };
+
+        window.XMLHttpRequest           = xmlHttpRequestWrapper;
+        xmlHttpRequestWrapper.prototype = xmlHttpRequestProto;
+        xmlHttpRequestWrapper.toString  = () => xmlHttpRequestToString;
+
+        // NOTE: We cannot just assign constructor property in OS X 10.11 safari 9.0
+        nativeMethods.objectDefineProperty.call(window.Object, xmlHttpRequestProto, 'constructor', {
+            value: xmlHttpRequestWrapper
+        });
+
+        const defineConstructorConst = (name, value) => {
+            nativeMethods.objectDefineProperty.call(window.Object, xmlHttpRequestWrapper, name, {
+                value,
+
+                enumerable: true
+            });
+        };
+
+        defineConstructorConst('UNSENT', 0);
```

Comment:
`defineProperty(xmlHttpRequestWrapper, 'UNSENT', XMLHttpRequest.UNSENT)`
and etc.

I propose this variant because it is more readable.

Also XMLHttpRequest has another static properties such as `name`, `valueOf` and etc.
But I think we can skip it.